### PR TITLE
[LLT-5366] Update requests dependency due to BDSA-2024-2985

### PR DESCRIPTION
**Pull request by Jauler**
### Problem
BDSA-2024-2985 was reported for psf-request, Update should be easy, so doing it in this PR.

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean ([requirements](../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

**Comment by Jauler** [comment]
This PR will also have to add a reference to new tag of libtelio-build.

**Comment by Jauler** [comment]
So after fixing ports to some non-ephemeral range, I realised that there is another problem, as seems like Pyro5 does not se SO_REUSEADDR or SO_REUSEPORT (well why should it without any additional configs), and in server shutdown may happen "in any direction" leaving TCP connection in those `TIME_WAIT` or state, which causes next tests to fail to open sockets. 

Will have to think about the approach, on solving this. Can we start using SO_REUSEADDR? :thinking: As it would be appropriate in this context.

**Review by tomaszklak — DISMISSED** [review]
+2

**Comment by stalowyjez** [comment]
+1

**Review by stalowyjez — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -15,4 +15,4 @@ libtelio-build-pipeline:
 
   trigger:
     project: $LIBTELIO_BUILD_PROJECT_PATH
-    branch: v1.2.5
+    branch: v1.2.5
```
Comment (stalowyjez) [inline .gitlab-ci.yml:18]: Why these newlines are removed?

Comment (Jauler) [reply .gitlab-ci.yml:18]: Oh it is a leftover after conflict resolution. I will fix it.

Comment (Jauler) [reply .gitlab-ci.yml]: fixed

**Review by stalowyjez — APPROVED** [review]
+2

**Pull request merged**